packages/frontend/src/components/chat/ChatListHelpers.tsx

```diff
@@ -121,3 +165,53 @@ export function useChatList(
     setQueryContactId,
   }
 }
+
+/**
+ * Despite the name, fetching is still debounced on ChatlistChanged events.
+ * (but it's not debounced when the hook's arguments change).
+ */
+function useChatListNoDebounce(
+  accountId: number,
+  listFlags: number | null = null,
+  queryStr: string = '',
+  queryContactId?: number
+) {
+  const [chatListEntries, setChatListEntries] = useState<number[]>([])
+
+  // Though perhaps a throttle would be more appropriate.
```

Comment:
as long is it is called at the falling edge (so in the end it is always updated) anything should be fine that reduces requests.

Reply:
Well, if we get this event at a sustained rate for a prolonged period of time, we'd only load the chat list at the end. It'd be better to load the chat list at least every second in such a case.

Reply:
good point. we need both:
- don't call it too often / overlapping
- call it latest every second
- call it at the end to make sure the correct state is shown in the end.

Maybe some code that waits for the request to complete (resolve or reject) before calling it again? But if core should not answer for any reason that could hang, so maybe just go for a classic time based throttle?

Reply:
Yes, I've been looking at [`async-throttle`](https://www.npmjs.com/package/@jcoreio/async-throttle), but it depends on Babel runtime. I didn't want to add bloat, or fork the package just yet as it will take some work.

But, prior to this MR, we already use `debounce`, so merging this MR wouldn't be a regression in this regard.
I'd suggest to tackle this later, because there are probably a few more places in the codebase where `throttle` makes more sense than `debounce`.

Reply:
>  fork the package just yet as it will take some work.

Looks easy enough (looking at the uncompiled `@jcoreio/async-throttle/index.js.flow` we'd only need to copy that file, add a license/attribution comment and convert it to TS. No need to fork the whole npm package.

Reply:
I tried, but it's not JS. It's something called "flow" XD